comp/core/workloadmeta/collectors/util/kubernetes_resource_parsers/pod.go

```diff
@@ -62,6 +63,20 @@ func (p podParser) Parse(obj interface{}) workloadmeta.Entity {
 		rtcName = *pod.Spec.RuntimeClassName
 	}
 
+	var gpuVendorList []string
+	uniqueGPUVendor := make(map[string]struct{})
+	for _, container := range pod.Spec.Containers {
+		for resourceName := range container.Resources.Limits {
+			gpuName, found := gpu.ExtractSimpleGPUName(gpu.ResourceGPU(resourceName))
+			if found {
+				uniqueGPUVendor[gpuName] = struct{}{}
+			}
+		}
+	}
+	for gpuVendor := range uniqueGPUVendor {
+		gpuVendorList = append(gpuVendorList, gpuVendor)
+	}
```

Comment:
Do we need this extra loop? 

Could we avoid the extra variable `uniqueGPUVendor`?

```go
var gpuVendorList []string
	
for _, container := range pod.Spec.Containers {
	for resourceName := range container.Resources.Limits {
		gpuName, found := gpu.ExtractSimpleGPUName(gpu.ResourceGPU(resourceName))
		if found {
			gpuVendorList = append(gpuVendorList, gpuName)
		}
	}
}
```

Reply:
I think I still want to have the `uniqueGPUVendor` to avoid duplication of multiple containers in a pod requesting the same gpu resource. However, I agree the extra for-loop can probably be removed. I plan on refactoring the way GPU resources are parsed/stored in a subsequent PR to have their own structs with additional metadata so all of this will be different soon. Thanks for pointing this out!